funcbench/env.go

```diff
@@ -128,19 +138,27 @@ func newGitHubEnv(ctx context.Context, e environment, gc *gitHubClient, workspac
 	return g, nil
 }
 
-func (g *GitHub) PostErr(ctx context.Context, err string) error {
-	if err := g.client.postComment(ctx, fmt.Sprintf("%v. Benchmark did not complete, please check action logs.", err)); err != nil {
+func (g *GitHub) PostErr(err string) error {
+	if err := g.client.postComment(fmt.Sprintf("%v. Benchmark did not complete, please check action logs.", err)); err != nil {
 		return errors.Wrap(err, "posting err")
 	}
 	return nil
 }
 
-func (g *GitHub) PostResults(ctx context.Context, cmps []BenchCmp, extraInfo ...string) error {
+func (g *GitHub) PostResults(tables []*benchstat.Table, extraInfo ...string) error {
 	b := bytes.Buffer{}
-	Render(&b, cmps, false, false, g.compareTarget)
+	if err := formatMarkdown(&b, tables); err != nil {
+		return err
+	}
+
 	legend := fmt.Sprintf("Old: `%s`\nNew: `PR-%d`", g.compareTarget, g.client.prNumber)
-	result := fmt.Sprintf("<details><summary>Click to check benchmark result</summary>\n\n%s\n%s\n%s</details>", legend, strings.Join(extraInfo, "\n"), formatCommentToMD(b.String()))
-	return g.client.postComment(ctx, result)
+	result := fmt.Sprintf(
+		"<details><summary>Click to check benchmark result</summary>\n\n%s\n%s\n%s</details>",
+		legend,
+		strings.Join(extraInfo, "\n"),
+		b.String(),
```

Comment:
For some reason the results are not getting posted in the results: 
https://github.com/prometheus-community/prometheus/pull/69#issuecomment-652632042

Reply:
I think this is a special case, you tested a benchmark existed in your PR, but not on branch master. I am testing against `BenchmarkRangeQuery` now, soon will post the benchmark result.

Reply:
Ah, so in that case if we actually ran funcbench on https://github.com/prometheus/prometheus/pull/7332  it would not show us the expected results. That's one of the usecase funcbench is supposed to be used for. So if there are new benchmarks added should it copy those to to the target branch/commit, ah i can already see this going little bit crazy. So unless, the PR is merged funcbench won't show expected results on https://github.com/prometheus/prometheus/pull/7332 , is that right?

cc: @krasi-georgiev

Reply:
also related to #398 , currently if there are no matching benchmarks, funcbench just runs and exits with 0.

Reply:
@nevill if we're not resolving this here, I think exiting with error if empty will be better than posting empty results.

Reply:
For sure, newly created benchmarks cannot have anything to compare with. We can post comments like `no matching benchmark to compare with` back.